src/provider/provider_level_zero.c

```diff
@@ -347,12 +354,28 @@ static umf_result_t ze_memory_provider_open_ipc_handle(void *provider,
     ASSERT(providerIpcData != NULL);
     ASSERT(ptr != NULL);
     ze_result_t ze_result;
-    ze_ipc_mem_handle_t *ze_ipc_handle = (ze_ipc_mem_handle_t *)providerIpcData;
+    ze_ipc_data_t *ze_ipc_data = (ze_ipc_data_t *)providerIpcData;
     struct ze_memory_provider_t *ze_provider =
         (struct ze_memory_provider_t *)provider;
+    int fd_local = -1;
+
+    if (ze_ipc_data->pid != utils_getpid()) {
+        int fd_remote = -1;
+        memcpy(&fd_remote, &ze_ipc_data->ze_handle, sizeof(fd_remote));
```

Comment:
Do the first `sizeof(int)` bytes of data of `ze_ipc_mem_handle_t` contain a file descriptor?

Reply:
Yes, it is a hack that MPI and oneCCL must use because the L0 IPC implementation is broken. Once L0 fixes it, it will not be needed.